test/gui/shared/steps/steps.py

```diff
@@ -239,6 +239,20 @@ def step(context, receiver, resource, permissions):
     openSharingDialog(context, resource)
     shareItem = SharingDialog()
     shareItem.addCollaborator(receiver, permissions)
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
+
+
+@When('the user adds following collaborators of resource "|any|" using the client-UI')
+def step(context, resource):
+    openSharingDialog(context, resource)
+    shareItem = SharingDialog()
+
+    for row in context.table[1:]:
+        receiver = row[0]
+        permissions = row[1]
+        shareItem.addCollaborator(receiver, permissions)
+
+    clickButton(waitForObject(shareItem.SHARING_DIALOG_CLOSE_BUTTON))
```

Comment:
since it is being used in many places, you should create a function, and try to reuse it.

Reply:
The reason for doing this is that we already have `SharingDialog` object so if we make a new function for this then every time we will have two object of `SharingDialog` instead fo having to do that I did this.